packages/messaging/ios/RNFBMessaging/RNFBMessagingModule.m

```diff
@@ -247,6 +247,7 @@ - (NSDictionary *)constantsToExport {
   if (@available(iOS 10.0, *)) {
     if ([UIApplication sharedApplication].isRegisteredForRemoteNotifications == YES) {
       resolve(@([RCTConvert BOOL:@(YES)]));
+      return;
```

Comment:
@mikehardy This return is logically inconsistent with the comment 5 lines later 
```
    // Apple docs recommend that registerForRemoteNotifications is always called on app start
    // regardless of current status
```

I don't know if the comment is accurate and I wasn't able to find any documentation from apple stating this when I just looked, but it is unclear to me now which cases we should call registerForRemoteNotifications.

Reply:
This was not intended to be any sort of a change that users should have to *alter* their code for. Whatever you have been doing, you should keep doing.

There have been other cases in this code base (and others I've been on) where resolving and then not returning has led to unintended side-effects as a resolve without return means code keeps executing.

In this case I judged the hygiene risk of that scenario much higher than attempting to register again, since by the conditional we are already registered.

That said, if you determine any case at all where your app does not seem to be registered, please let me know. That would be entirely unexpected (given the conditional being true in this case....)

Reply:
reading a bit further, it looks like the full/complete implementation of this Apple API would be implementing didRegisterForRemoteNotificationsWithDeviceToken and didFailToRegisterForRemoteNotificationsWithError and the resolving the promise with the result of them being called.

https://developer.apple.com/documentation/appkit/nsapplicationdelegate/1428766-application

I've found nothing yet to indicate if the app should always call registerForRemoteNotifications or if it can count on isRegisteredForRemoteNotifications and only call it registerForRemoteNotifications when it is false.

Reply:
Just noticed I'm using my personal account, this is all from brian@slashtalk.com, my mistake on not checking the browser tab.

Reply:
> reading a bit further, it looks like the full/complete implementation of this Apple API would be implementing didRegisterForRemoteNotificationsWithDeviceToken and didFailToRegisterForRemoteNotificationsWithError and the resolving the promise with the result of them being called.
> 
> https://developer.apple.com/documentation/appkit/nsapplicationdelegate/1428766-application

That's a macOS API (`NSApplication` root).

I think  you'll like that when we register we store the promise resolver and rejecter on our AppDelegate object, and we've swizzled ourselves into the response chain for when certain items are invoked, and we do what you mention, but on the `UIApplication` iOS APis:

https://github.com/invertase/react-native-firebase/blob/8765a993f137964f11f2dffbb027d65cda897204/packages/messaging/ios/RNFBMessaging/RNFBMessaging%2BAppDelegate.m#L90-L115

> 
> I've found nothing yet to indicate if the app should always call registerForRemoteNotifications or if it can count on isRegisteredForRemoteNotifications and only call it registerForRemoteNotifications when it is false.

I have not either. It's actually called immediately if you're set to auto register for them 

https://github.com/invertase/react-native-firebase/blob/5daa5ff7f22aed5de338d1d1ca5bb430cc93f8c5/packages/app/firebase-schema.json#L81-L83

https://github.com/invertase/react-native-firebase/blob/8765a993f137964f11f2dffbb027d65cda897204/packages/messaging/ios/RNFBMessaging/RNFBMessaging%2BNSNotificationCenter.m#L91-L93

I'm unaware of who you are as @ourhut or `brian@slashtalk.com`, but if there is something wrong here, I can certainly correct it

Reply:
Thank you for pointing out the place the location the promise resolver and rejecter are referenced, I was looking around and hadn't found it yet. The makes perfect sense. 

I also now see why what you've done is correct, and I don't have any cause for concern other than what I got from reading the comment I mentioned.

A style comment that I think would make this clearer, move call to registerForRemoteNotifications into the else block after registering the resolvers, and remove the early return and old comment. I feel like that makes it clear the there are two distinct paths, one that returns TRUE, and one that registers the resolvers and dispatches a request that will resolve/reject them later. Logically the same as what you have now. 

Thanks for quickly responding, and I apologize for the noise. This library is amazing, we're getting a lot of value from it and I'm happy to help with improving it as I can.

Reply:
Not sure if I'll have time to get to that suggestion right now, but I do agree things could be tidied up there.
Right now I'm still doggedly pursuing other initialization / config-setting changes that cross-cut the whole set of modules, there's quite a few things going on there and some of them have higher priority ecosystem impacts all the way back to firebase-ios-sdk like #4241 which I need to maintain focus on

Reply:
Oh, and no worries about noise - I was moving literally as quickly as possible through things over the weekend and no one else has had time to review, so even if the result is the null hypothesis is upheld (no further change necessary) simply having anyone look at it, and even potentially forcing a defense of change, is hugely helpful. I appreciate it